tools/autograd/gen_python_functions.py

```diff
@@ -80,6 +80,10 @@
   ${unpack_self}
   ParsedArgs<${max_args}> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
+  if (r.signature.deprecated) {
+    TORCH_WARN("This signature for ", r.signature.name, " is deprecated:\n",
+        "${name}", r.signature.toString());
```

Comment:
nit: it is best if in deprecation messages you can say what the correct thing to use instead is. Unfortunately, this is fairly generic code so I'm not sure if there's actually a way to conveniently get that information here, but it's worth keeping an eye out on. (Maybe this message could print all the other, non-deprecated overloads)

Reply:
Also, signature is implementation internal nomenclature; it's probably better to refer to this as overload for users